semgrep_output_v1.atd

```diff
@@ -1647,7 +1647,12 @@ type ci_scan_complete = {
   ?final_attempt: bool option; (* always optional *)
   }
 
+(* Update the stats_schema_version if you are changing this type.
```

Comment:
I'm not sure I understand the goal of this schema_version; We can't have breaking change ... we must support older versions of Semgrep, which is why for instance you've added this new field as a `?` because otherwise we would not parse correctly the output of semgrep 1.95.0 for example.
We have actually a window of versions we support (@chmccreery is actually working on moving up the minimum version to 1.50.0 or something) but in any case what matters is this mimimal version of semgrep. So we already send the semgrep version in another field; we don't need another version scheme.

Reply:
We can’t have breaking changes — that’s totally clear. The schema version idea is more about making life easier for downstream tasks, especially in Snowflake.

Right now, the stats JSON gets transformed into a stats table. If we ever evolve the schema — like adding fields —it’d be really handy to have a version marker that tells Snowflake tasks, “Hey, this is the structure you’re dealing with.” We can definitely check for the existence of new keys instead, but having a schema version simplifies things a lot, especially when working with historical data. 

For the Data Science team, it means they don’t have to guess when fields were added or rework transformation - it’s just cleaner and easier to manage. It’s kind of a future-proofing move. The schema evolves, nothing breaks, and downstream logic stays sane. 

That said, now that I think about it, perhaps when doing the data science bit, we can join with the scan table and grab the semgrep_version from the meta column. We could use semgrep_version in place of this proposed schema version. Or maybe just bake the semgrep verison into stats? What do you think?

Reply:
Yeah, I like using the `semgrep version` for this, but don't feel too strongly. It just seems like another thing to try to keep in sync manually, so another opportunity for human-introduced errors.